server/services/MDSEnabledClientService.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  ILegacyCustomClusterClient,
+  OpenSearchDashboardsRequest,
+} from 'opensearch-dashboards/server';
+import { RequestHandlerContext } from '../../../../src/core/server';
+
+export abstract class MDSEnabledClientService {
+  constructor(private osDriver: ILegacyCustomClusterClient, private dataSourceEnabled: boolean) {}
+
+  protected getClient(request: OpenSearchDashboardsRequest, context: RequestHandlerContext) {
+    const dataSourceId = (request.query as any).dataSourceId;
```

Comment:
`(request.query as any)?.dataSourceId`;

Reply:
The query object will always be there since it is added by the browser (it could be empty if no query param is provided in the request)